lib/handler/redirect.js

```diff
@@ -1,11 +1,29 @@
 'use strict'
 
 const util = require('../core/util')
+const { kBodyUsed } = require('../core/symbols')
 const assert = require('assert')
 const { InvalidArgumentError } = require('../core/errors')
 
 const redirectableStatusCodes = [300, 301, 302, 303, 307, 308]
 
+const kBody = Symbol('body')
+
+class BodyAsyncIterable {
+  constructor (body) {
+    this[kBody] = body
+    this[kBodyUsed] = false
+  }
+
+  async * [Symbol.asyncIterator] () {
+    if (this[kBodyUsed]) {
+      throw new TypeError('disturbed')
```

Comment:
I would throw a more meaningful error, this one looks a little obscure.

Reply:
I'll change it to an assertion. It shouldn't happen.